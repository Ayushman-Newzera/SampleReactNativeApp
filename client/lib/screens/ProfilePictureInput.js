var _interopRequireDefault = require('@babel/runtime/helpers/interopRequireDefault');
Object.defineProperty(exports, '__esModule', {value: true});
exports.default = void 0;
var _react = _interopRequireDefault(require('react'));
var _reactNative = require('react-native');
var _SimpleImagePicker = _interopRequireDefault(require('./SimpleImagePicker'));
var _jsxFileName =
  '/Users/ayushman/Desktop/SampleReactNativeApp1/client/src/screens/ProfilePictureInput.js';
function ProfilePictureInput(_ref) {
  var route = _ref.route,
    navigation = _ref.navigation;
  var handleNavigation = function handleNavigation() {
    navigation.navigate('Home');
  };
  return _react.default.createElement(
    _reactNative.View,
    {
      style: styles.container,
      __self: this,
      __source: {fileName: _jsxFileName, lineNumber: 13, columnNumber: 5},
    },
    _react.default.createElement(_SimpleImagePicker.default, {
      asyncKey: '@profilePicture',
      handleNavigation: handleNavigation,
      __self: this,
      __source: {fileName: _jsxFileName, lineNumber: 14, columnNumber: 7},
    }),
  );
}
var styles = _reactNative.StyleSheet.create({
  container: {position: 'relative', flex: 1},
});
var _default = ProfilePictureInput;
exports.default = _default;
