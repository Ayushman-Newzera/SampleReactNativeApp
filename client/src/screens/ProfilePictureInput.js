// @flow

import React from 'react';
import {StyleSheet, View} from 'react-native';
import SimpleImagePicker from './SimpleImagePicker';

function ProfilePictureInput({route, navigation}) {
  const handleNavigation = () => {
    navigation.navigate('Home');
  };

  return (
    <View style={styles.container}>
      <SimpleImagePicker
        asyncKey="@profilePicture"
        handleNavigation={handleNavigation}
      />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    position: 'relative',
    flex: 1,
  },
});

export default ProfilePictureInput;
