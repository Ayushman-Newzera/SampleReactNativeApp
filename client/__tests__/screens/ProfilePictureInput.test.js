import {shallow} from 'enzyme';
import React from 'react';
import ProfilePictureInput from '../../src/screens/ProfilePictureInput';

const createTestProps = props => ({
  navigation: {
    navigate: jest.fn(),
  },
  ...props,
});

describe('UselessTextInputMultiline', () => {
  it('TextInput should render correctly', () => {
    const component = shallow(<ProfilePictureInput />);
    expect(component.exists()).toBe(true);
    expect(component).toMatchSnapshot();
  });

  describe('HandleNavigation function should work correctly', () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = createTestProps({});
      wrapper = shallow(<ProfilePictureInput {...props} />);
    });

    it('HandleNavigation function should work correctly', async () => {
      const SimpleImagePicker = wrapper.find('SimpleImagePicker').get(0);
      await SimpleImagePicker.props.handleNavigation();
      expect(props.navigation.navigate).toHaveBeenCalledWith('Home');
    });
  });
});
