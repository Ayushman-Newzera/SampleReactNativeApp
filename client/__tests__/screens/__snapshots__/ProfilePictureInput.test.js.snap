// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UselessTextInputMultiline TextInput should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "position": "relative",
    }
  }
>
  <SimpleImagePicker
    asyncKey="@profilePicture"
    handleNavigation={[Function]}
  />
</View>
`;
